packages/affiliates/gas-rebate/VoterGasRebateV2.ts

```diff
@@ -0,0 +1,128 @@
+// This is a script calculates the gas rebate for UMA protocol voters. The script finds all VoteCommitted and
+// VoteRevealed events in a specified time range and calculates the gas used for each event. It then uses UniswapV3 swap
+// events to find the UMA/ETH price at the block in which each event was included and calculates the ETH cost of the gas
+// used in UMA tokens. The script aggregates the gas rebates by voter and saves the results to a file. // It is designed
+// to not require any run time parameters by always running it one month after the desired output month.
+
+import { getAddress } from "@uma/contracts-node";
+const hre = require("hardhat");
+const { ethers } = hre;
+import { BigNumber } from "ethers";
+import bn from "bignumber.js";
+const moment = require("moment");
+import { findBlockNumberAtTimestamp, getWeb3, decodePriceSqrt } from "@uma/common";
+import fs from "fs";
+import path from "path";
+
+// Find the closest UMA/ETH price to the given block number by searching through swap events in the given range.
+function findClosestUmaEthPrice(swapEvents: any, blockNumber: number) {
+  const closest = swapEvents.reduce((prev: any, curr: any) => {
+    return Math.abs(curr.blockNumber - blockNumber) < Math.abs(prev.blockNumber - blockNumber) ? curr : prev;
+  });
+  // The returned price is of the form 1 UMA = x ETH. We want to return the price of 1 ETH in UMA. Also, Uniswap returns
+  // the price in x96 square root encoded. We need to decode by reverse x96 operation by compting (price/(2^96))^2.
+  const basePrice = decodePriceSqrt(closest.args.sqrtPriceX96.toString());
+  const bnPrice = new bn(ethers.utils.parseUnits("1").toString()).div(new bn(basePrice.toString()));
+  return BigNumber.from(bnPrice.toString().toString().substring(0, bnPrice.toString().indexOf(".")));
```

Comment:
its a little weird to convert here between bignumberjs and bn, maybe this is just to avoid integer math

Reply:
ye, the reason is that `decodePriceSqrt` returns a `bn` not `BigNumber` its a bit janky but I force it back to the base case as quickly as posible.